tests/integration/test_charm.py

```diff
@@ -332,6 +337,215 @@ async def test_remove_postgresql_metastore_relation(ops_test: OpsTest, test_pod,
+    print(process.stderr.decode())
+    logger.info(f"JDBC endpoint test returned with status {process.returncode}")
+    assert process.returncode == 1
+    assert "Error validating the login" in process.stderr.decode()
+
+
+@pytest.mark.abort_on_fail
+async def test_jdbc_endpoint_valid_credentials(ops_test: OpsTest, test_pod):
+    """Test the JDBC connection when invalid credentials are provided."""
+    logger.info("Running action 'get-jdbc-endpoint' on kyuubi-k8s unit...")
+    kyuubi_unit = ops_test.model.applications[APP_NAME].units[0]
+    action = await kyuubi_unit.run_action(
+        action_name="get-jdbc-endpoint",
+    )
+    result = await action.wait()
+
+    jdbc_endpoint = result.results.get("endpoint")
+    logger.info(f"JDBC endpoint: {jdbc_endpoint}")
+
+    logger.info("Running action 'get-password' on kyuubi-k8s unit...")
+    action = await kyuubi_unit.run_action(
+        action_name="get-password",
+    )
+    result = await action.wait()
+
+    password = result.results.get("password")
+    logger.info(f"Fetched password: {password}")
+
+    username = "admin"
+
+    logger.info(
+        f"Testing JDBC endpoint by connecting with beeline with username={username} and password={password} ..."
+    )
+    process = subprocess.run(
+        [
+            "./tests/integration/test_jdbc_endpoint.sh",
+            test_pod,
+            jdbc_endpoint,
+            "db_333",
+            "table_333",
+            username,
+            password,
+        ],
+        capture_output=True,
+    )
+    print("========== test_jdbc_endpoint.sh STDOUT =================")
+    print(process.stdout.decode())
+    print("========== test_jdbc_endpoint.sh STDERR =================")
+    print(process.stderr.decode())
+    logger.info(f"JDBC endpoint test returned with status {process.returncode}")
+    assert process.returncode == 0
+
+
+@pytest.mark.abort_on_fail
+async def test_set_password_action(ops_test: OpsTest):
+    """Test set-password action."""
+    logger.info("Running action 'get-password' on kyuubi-k8s unit...")
+    kyuubi_unit = ops_test.model.applications[APP_NAME].units[0]
+    action = await kyuubi_unit.run_action(
+        action_name="get-password",
+    )
+    result = await action.wait()
+    old_password = result.results.get("password")
+
+    logger.info("Running action 'set-password' on kyuubi-k8s unit...")
+    password_to_set = str(uuid.uuid4())
+    action = await kyuubi_unit.run_action(action_name="set-password", password=password_to_set)
+    result = await action.wait()
+    assert result.results.get("password") == password_to_set
+
+    logger.info("Running action 'get-password' on kyuubi-k8s unit...")
+    action = await kyuubi_unit.run_action(
+        action_name="get-password",
+    )
+    result = await action.wait()
+    new_password = result.results.get("password")
+
+    assert new_password != old_password
+    assert new_password == password_to_set
```

Comment:
After the change of password we need to check if the connection will succeed with the new password

Reply:
Updated.